package.json

```diff
@@ -39,6 +39,7 @@
   "dependencies": {
     "@google-cloud/common": "^0.12.0",
     "continuation-local-storage": "^3.1.4",
+    "extend": "^3.0.0",
```

Comment:
We need to use the deep copying capability for nested objects in the config: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign#Deep_Clone.